### ISSUE TITLE
Testing problem

### ISSUE THREAD

**Issue by garrysadd** [zopefoundation/Products.PythonScripts#18]
Logs you out when testing a script with default content. Blank page returned after login box dismissed.

**Comment by icemac** [comment]
I will look into this issue on the next Zope sprint in Halle in two weeks.

**Comment by icemac** [comment]
@garrysadd Which version of Python, Zope, RestrictedPython and Products.PythonScripts do you use?

On the current `master` of Zope, RestrictedPython, and Products.PythonScripts on Python 3.6 the test tab returns on my machine: `This is the Script (Python) "py-default" in http://localhost:8080`

**Comment by garrysadd** [comment]
The Python scripts installed via this command "pip install Products.PythonScripts"
The rest as below:
AccessControl==4.0b4
Acquisition==4.4.4
AuthEncoding==4.0.0
beautifulsoup4==4.6.3
BTrees==4.5.0
Chameleon==3.2
DateTime==4.2
DocumentTemplate==3.0b3
ExtensionClass==4.3.0
five.globalrequest==99.1
five.localsitemanager==3.1
funcsigs==1.0.2
futures==3.2.0
ipaddress==1.0.22
Missing==4.0.1
mock==2.0.0
MultiMapping==4.0
PasteDeploy==1.5.2
pbr==4.0.3
Persistence==3.0b3
persistent==4.2.4.2
Products.BTreeFolder2==4.0.0
Products.PythonScripts==4.1
Products.ZCatalog==4.1
Products.ZCTextIndex==4.0.2
Products.ZSQLMethods==3.0.3
psycopg2==2.7.5
python-gettext==3.0
pytz==2018.4
Record==3.4
RestrictedPython==4.0b4
six==1.11.0
SQLAlchemy==1.2.11
transaction==2.2.1
trollius==2.2
waitress==1.1.0
WebOb==1.8.1
WebTest==2.0.29
WSGIProxy2==0.4.4
z3c.pt==3.1.0
z3c.sqlalchemy==1.4.0
zc.lockfile==1.3.0
ZConfig==3.2.0
zdaemon==4.2.0
ZEO==5.2.0
zExceptions==4.0
ZODB==5.4.0
zodbpickle==1.0.1
-e git+https://github.com/zopefoundation/Zope.git@c9002f06a5a94841d2cbf6b51d366d3b7f633297#egg=Zope
zope.annotation==4.6.0
zope.browser==2.2.0
zope.browsermenu==4.3.0
zope.browserpage==4.2.0
zope.browserresource==4.2.1
zope.cachedescriptors==4.3.1
zope.component==4.4.1
zope.componentvocabulary==2.1.0
zope.configuration==4.1.0
zope.container==4.2.1
zope.contentprovider==4.1.0
zope.contenttype==4.3.0
zope.datetime==4.2.0
zope.deferredimport==4.2.1
zope.deprecation==4.3.0
zope.dottedname==4.2
zope.event==4.3.0
zope.exceptions==4.2.0
zope.filerepresentation==4.2.0
zope.formlib==4.4
zope.globalrequest==1.4
zope.i18n==4.3.1
zope.i18nmessageid==4.1.0
zope.interface==4.5.0
zope.lifecycleevent==4.2.0
zope.location==4.1.0
zope.pagetemplate==4.3.0
zope.processlifetime==2.2.0
zope.proxy==4.3.0
zope.ptresource==4.1.0
zope.publisher==4.3.2
zope.ramcache==2.2.0
zope.schema==4.5.0
zope.security==4.2.2
zope.sendmail==4.1.0
zope.sequencesort==4.0.1
zope.site==4.1.0
zope.size==4.2.0
zope.sqlalchemy==1.0
zope.structuredtext==4.2.0
zope.tal==4.3.1
zope.tales==4.2.0
zope.testbrowser==5.2.4
zope.testing==4.6.2
zope.testrunner==4.8.1
zope.traversing==4.2.0
zope.viewlet==4.1.0
Zope2==4.0b1
ZServer==4.0b1

**Comment by icemac** [comment]
@garrysadd Does the error happen when you call the test URL as authorised user or as anonymous?

**Comment by garrysadd** [comment]
As manager in the ZMI. If you delete the default content then it will return a print statement like "return 'Hello'".
Look like something in the default content is unauthorised, but can't say what as there is no error returned, just a blank screen. The error log is of no use as that seems to be broken too.

**Comment by icemac** [comment]
I tried the following steps using Python 3.6:

```bash
$ python3.6 -m venv zope
$ cd zope
$ bin/pip install Zope==4.0b7 -c https://zopefoundation.github.io/Zope/releases/4.0b7/constraints.txt
$ bin/pip install Products.PythonScripts
$ bin/mkwsgiinstance -d .
$ bin/runwsgi etc/zope.ini
```

In the browser I created a `Script (Python)` and clicked on the `Test` tab. It returns the text as generated by the script. I neither get an error message nor a basic auth dialog.
(I even tried with an id containing a non-ASCII character.)

The same happens on Python 2.7.

Are you able to confirm that the error does not happen with Zope 4.0b7?
